### ISSUE TITLE
Topics for Pier Luigi's session

### ISSUE THREAD

**Issue by surbhigoel77** [Cambridge-ICCS/practical-ml-with-pytorch#60]
Pier Luigi's + Anna Sommer:

Summary:

- Practical part will include 3 examples from the paper with a reduced dataset. 
- Students are required to be familiar with data visualisation ahead of time. Specifically `cartopy` and `matplotlib`. I have checked with James and this will be part of the visualisation lecture.
- Mostly using pandas.

Jack inquired about what we could teach to complement the follow session by Anna Sommer::

Plan is to do the following: 

I’m preparing a practical part for our session now. It’ll be in a form of Jupyter Notebook with some theoretical explanation, questions, and code to run. The topic is “Observation System Simulation Experiences in the Atlantic Ocean for enhanced surface ocean pCO2 reconstructions”, it is based on my paper with the same title published in 2021 [https://os.copernicus.org/articles/17/1011/2021/](https://eur03.safelinks.protection.outlook.com/?url=https%3A%2F%2Fos.copernicus.org%2Farticles%2F17%2F1011%2F2021%2F&data=05%7C02%7Cma595%40universityofcambridgecloud.onmicrosoft.com%7Ce45ad7a19acb4f43a13408dc893baa4d%7C49a50445bdfa4b79ade3547b4f3986e9%7C1%7C0%7C638536136579140421%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C0%7C%7C%7C&sdata=NeIssFbQtTY8PFZtbLKRn62TlH%2BJp2nddh7R%2Be04EQM%3D&reserved=0).
The practical part will contain three examples from this paper with reduced data set due to time restriction.
We would like to show students that physical models can be also used to plan a deployment of measuring instruments, particularly localizing the areas where the source of data is most important to understand some phenomena.  
 
The packages I use in the exercise are

```
import torch
import torch.nn as nn
import torch.optim as optim
from torch.utils.data import DataLoader, TensorDataset
from sklearn.model_selection import train_test_split
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import cartopy.crs as ccrs
 ```
 
I’m working with data in `dataframe` format. Data will be in `.csv` files.
An important part is a **data visualisation**.  It would be great if students have experienced already with **matplotlib** and **cartopy**.
- First, we will plot the distribution of available data in the Atlantic Ocean.
- In the second step, we’ll run Machine Learning model and to look at the loss function.
- The third part will be a visualisation of results: maps of standard deviation, mean of differences between reference (ocean model) and ML output, correlation coefficient between reference and ML output.

It is three main steps of the exercise with questions and analysing work in each of them.
 
If you need more details or have comments/questions/suggestions, please don’t hesitate to contact me.
 
Hi, Laura, Pier Luigi, mentioned that you could help during the practical part. As we have something around an hour for practical part it would be great to have someone to help if students struggle technically or have questions etc. If you feel comfortable with this topic it would be helpful if you join us. As mentioned before if you need more details or have comments/questions/suggestions, please don’t hesitate to contact me. I’ll share the Jyputer Notebook file when it is ready (hopefully in a week's time).

**Comment by surbhigoel77** [comment]
Pier Luigi's session requirements:

### Module imports:

import torch
import torch.nn as nn
import torch.optim as optim
from torch.utils.data import DataLoader, TensorDataset
from sklearn.model_selection import train_test_split
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import cartopy.crs as ccrs

### Tasks:
1. Load data from csv files
2. Dataframe of the data
3. Plot distribution of the variables
4. Training and Loss function
5. Visualisation using matplotlib and cartopy (maps of standard deviation, mean of differences between reference (ocean model) and ML output, correlation coefficient between reference and ML output.)
